twilio-authenticator-sample/src/sharedTest/java/com/twilio/authenticatorsample/mocks/MockTwilioAuthenticator.java

```diff
@@ -87,22 +87,29 @@ public void clearLocalData() {
     }
 
     @Override
-    public void addObserver(AuthenticatorObserver observer) {
-        observers.add(observer);
-        notifyObservers();
+    public void setMultiAppListener(@NonNull MultiAppListener multiAppListener) {
+        this.multiAppListener = multiAppListener;
+        notifyListener();
     }
 
-    private void notifyObservers() {
-        for(AuthenticatorObserver observer : observers) {
-            observer.onNewCode(apps);
-        }
+    @Override
+    public void setSingleAppListener(@NonNull SingleAppListener singleAppListener) {
+        this.singleAppListener = singleAppListener;
+        notifyListener();
     }
 
-    @Override
-    public void removeObserver(AuthenticatorObserver observer) {
-        observers.remove(observer);
+
+    private void notifyListener() {
+        if (multiAppListener != null) {
+            multiAppListener.onNewCode(apps);
+        }
+
+        if (singleAppListener != null) {
+            singleAppListener.onNewCode(apps.get(0));
```

Comment:
what happens if the array doesnt have any apps?

Reply:
It's a Mock. I'm guessing that scenario is not part of the tests yet ;)
Otherwise, it will hopefully make some tests fail

Reply:
Yes, it is a mock, but I added a validation to prevent errors